// @v2-lib/vue.spa.plugin/router/index.js

// import router from '@/router'
import _import from '../util/_import_production'

// console.log(router);
let menus=[];
let menuRoutes=[];

function filterAsyncRouter(asyncRouterMap) {

    const accessedRouters = asyncRouterMap.map(route => {

      if(!route.component){
        route.component=()=>import('../components/empty.vue');
      }
      
      if(typeof route.component === 'string') {
        route.componentUrl=route.component;
        route.component=_import(route.component)
  
      }
      if (route.children && route.children.length) {
        route.children = filterAsyncRouter(route.children)
      }
      return  route;
    })
  
    return accessedRouters;
  }
  
export  function getMenuRoutes(){
  return menuRoutes;
}
export function getMenus(){
  return menus; 
}
export function addMenus(data){
    menus=data;
}

export function addMenuToRoutes(data){
  let routes=window.router.options.routes;

  let menuRoute = filterAsyncRouter(data);
  menuRoutes.push(...menuRoute);
  routes[0].children.push(...data)
  window.router.addRoutes(routes);
}



export function addTabsRoutes(obj){
  let routes=window.router.options.routes;

  routes[0].children.push(obj)
  window.router.addRoutes(routes);

}


export function addMainRoutes(obj){

  let routes=window.router.options.routes;

   routes.push(obj)
   window.router.addRoutes(routes);
}


export function pathToRoutes(){
 
  let wpath = window.location.hash.split('?')[0].replace('#/', '');
  let inMenus=menus.filter(item=>item.path===wpath);
  if(inMenus && inMenus.length){
    addMenuToRoutes(inMenus);
  }
}

export default {
  getMenuRoutes,addMenuToRoutes,addTabsRoutes,addMainRoutes,getMenus,addMenus,pathToRoutes
}
